src/output-json-tls.c

```diff
@@ -74,13 +74,20 @@ typedef struct JsonTlsLogThread_ {
 void JsonTlsLogJSONBasic(json_t *js, SSLState *ssl_state)
 {
     /* tls.subject */
-    json_object_set_new(js, "subject",
-                        json_string(ssl_state->server_connp.cert0_subject));
+    if (ssl_state->server_connp.cert0_subject != NULL) {
+        json_object_set_new(js, "subject",
+                            json_string(ssl_state->server_connp.cert0_subject));
+    }
 
     /* tls.issuerdn */
-    json_object_set_new(js, "issuerdn",
-                        json_string(ssl_state->server_connp.cert0_issuerdn));
+    if (ssl_state->server_connp.cert0_subject != NULL) {
+        json_object_set_new(js, "issuerdn",
+                            json_string(ssl_state->server_connp.cert0_issuerdn));
+    }
 
+    if ((ssl_state->flags & SSL_AL_FLAG_SESSION_RESUMED) != 0) {
+        json_object_set_new(js, "session_resumed", json_true());
```

Comment:
I think something like "session":"resumed" would look better, even if it's a string an not a boolean.